fury/gltf.py

```diff
@@ -398,3 +399,404 @@ def load_camera(self, camera_id, transform_mat):
+        np_im = numpy_support.vtk_to_numpy(scalars)
+        np_im = np.reshape(np_im, (rows, cols, -1))
+
+        img = Image.fromarray(np_im)
+        image_path = f'{name}BaseColorTexture.png'
+        img.save(image_path)
+        write_material(gltf, 0, image_path)
+
+    if colors is not None:
+        ctype = comp_type.get(gltflib.FLOAT)
+        atype = acc_type.get(gltflib.VEC3)
+
+        shape = colors.shape[0]
+        colors = np.concatenate((colors, np.full((shape, 1), 255.)), axis=1)
+        colors = colors / 255
+        colors = colors.reshape((-1, )).astype(ctype['dtype'])
+        blength = len(colors)*ctype['size']
+        buff_file.write(colors.tobytes())
+        write_bufferview(gltf, 0, boffset, blength)
+        write_accessor(gltf, count, 0, gltflib.FLOAT, shape, gltflib.VEC4)
+        boffset += blength
+        color = count
+        count += 1
+    material = None if tcoords is None else 0
+    prim = get_prim(vertex, index, color, tcoord, normal, material)
+    return prim, boffset, count
+
+
+def write_scene(gltf, nodes):
+    """Create scene
+
+    Parameters
+    ----------
+    gltf : GLTF2
+        Pygltflib GLTF2 object
+    nodes : list
+        List of node indices.
+    """
+    scene = gltflib.Scene()
+    scene.nodes = nodes
+    gltf.scenes.append(scene)
+
+
+def write_node(gltf, mesh=None, camera=None):
+    """Create node
+
+    Parameters
+    ----------
+    gltf : GLTF2
+        Pygltflib GLTF2 object
+    mesh : int, optional
+        Mesh index
+    camera : int, optional
+        Camera index.
+    """
+    node = gltflib.Node()
+    if mesh is not None:
+        node.mesh = mesh
+    if camera is not None:
+        node.camera = camera
+    gltf.nodes.append(node)
+
+
+def write_mesh(gltf, primitives):
+    """Create mesh and add primitive.
+
+    Parameters
+    ----------
+    gltf : GLTF2
+        Pygltflib GLTF2 object.
+    primitives : list
+        List of Primitive object.
+    """
+    mesh = gltflib.Mesh()
+    for prim in primitives:
+        mesh.primitives.append(prim)
+
+    gltf.meshes.append(mesh)
+
+
+def write_camera(gltf, camera):
+    """Create and add camera.
+
+    Parameters
+    ----------
+    gltf : GLTF2
+        Pygltflib GLTF2 object.
+    camera : vtkCamera
+        scene camera.
+    """
+    orthographic = camera.GetParallelProjection()
+    cam = gltflib.Camera()
+    if orthographic:
+        cam.type = "orthographic"
+    else:
+        clip_range = camera.GetClippingRange()
+        angle = camera.GetViewAngle()
+        ratio = camera.GetExplicitAspectRatio()
+        aspect_ratio = ratio if ratio else 1.0
+        pers = gltflib.Perspective()
+        pers.aspectRatio = aspect_ratio
+        pers.znear, pers.zfar = clip_range
+        pers.yfov = angle * np.pi/180
+        cam.type = "perspective"
+        cam.perspective = pers
+    gltf.cameras.append(cam)
+
+
+def get_prim(verts, indices, cols, tcoords, normals, mat):
```

Comment:
Maybe you should try using full terms for eg. `vertex`, `color`, `material`, or something similar depending on the value they have. I am also a bit confused because you have written `verts` (plural) but as per the docs they store `int` (single value).